lib/Runtime/Library/AtomicsObject.cpp

```diff
@@ -0,0 +1,270 @@
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_NeedSharedArrayBufferObject);
+        }
+
+        return arrayBuffer;
+    }
+
+    uint32 AtomicsObject::ValidateAtomicAccess(Var typedArray, Var requestIndex, ScriptContext *scriptContext)
+    {
+        int32 accessIndex = -1;
+        if (TaggedInt::Is(requestIndex))
+        {
+            accessIndex = TaggedInt::ToInt32(requestIndex);
+        }
+        else
+        {
+            accessIndex = JavascriptConversion::ToInt32_Full(requestIndex, scriptContext);
+            double dblValue = JavascriptConversion::ToNumber(requestIndex, scriptContext);
+            if (dblValue != accessIndex)
+            {
+                JavascriptError::ThrowRangeError(scriptContext, JSERR_InvalidTypedArrayIndex);
+            }
+        }
+
+        Assert(TypedArrayBase::Is(typedArray));
+        
+        if (accessIndex < 0 || accessIndex >= (int32)TypedArrayBase::FromVar(typedArray)->GetLength())
+        {
+            JavascriptError::ThrowRangeError(scriptContext, JSERR_InvalidTypedArrayIndex);
+        }
+
+        return (uint32)accessIndex;
+    }
+
+    TypedArrayBase * AtomicsObject::ValidateAndGetTypedArray(Var typedArray, Var index, __out uint32 *accessIndex, ScriptContext *scriptContext, bool onlyInt32)
+    {
+        ValidateSharedIntegerTypedArray(typedArray, scriptContext, onlyInt32);
+        uint32 i = ValidateAtomicAccess(typedArray, index, scriptContext);
+        if (accessIndex != nullptr)
+        {
+            *accessIndex = i;
+        }
+        Assert(TypedArrayBase::Is(typedArray));
+        return TypedArrayBase::FromVar(typedArray);
+    }
+
+    Var AtomicsObject::EntryAdd(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.add");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedAdd(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryAnd(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(3, "Atomics.and");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+        return typedArrayBase->TypedAnd(accessIndex, args[3]);
+    }
+
+    Var AtomicsObject::EntryCompareExchange(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ATOMICS_FUNCTION_BEGIN_WRAPPER(4, "Atomics.compareExchange");
+
+        uint32 accessIndex = 0;
+        TypedArrayBase *typedArrayBase = ValidateAndGetTypedArray(args[1], args[2], &accessIndex, scriptContext);
+
+        Var comparand = args[3];
+        Var replacementValue = args[4];
+
+        return typedArrayBase->TypedCompareExchange(accessIndex, comparand, replacementValue);
```

Comment:
>return typedArrayBase->TypedCompareExchange(accessIndex, comparand, replacementValue); [](start = 8, length = 86)

Why create local variables for args[3] and args[4], pass directly (as done elsewhere in this file) for consistency.